docs/registering-strategies.adoc

```diff
@@ -0,0 +1,80 @@
+
+=== Implementing `TestExecutionPlannerFactory`
+
+`TestExecutionPlannerFactory` is a Java SPI interface that manages the creation of the strategy.
+You need to implement three simple methods, two related with the name of the strategy and one that generates the strategy class.
+
+As an example from `failed` strategy:
+
+[source, java]
+.FailedTestsDetectorFactory.java
+----
+include::../strategies/failed/src/main/java/org/arquillian/smart/testing/strategies/failed/FailedTestsDetectorFactory.java[tag=documentation]
+----
+<1> Returns the alias of the strategy
+<2> Returns an instance of `TestExecutionPlanner`
+
+Of course as any other Java SPI, you need to register it in `META-INF/services`.
+
+[source, properties]
+.META-INF/services/org.arquillian.smart.testing.spi.TestExecutionPlannerFactory
+----
+include::../strategies/failed/src/main/resources/META-INF/services/org.arquillian.smart.testing.spi.TestExecutionPlannerFactory[]
+----
+
+=== Implementing `TestExecutionPlanner`
+
+`TestExecutionPlanner` is the strategy implementation of the strategy.
+The important method there is `getTests()` which returns a collection of the tests that must be executed.
+You also need to set there the strategy name.
+
+For example, failed strategy is implemented as:
+
+[source, java]
+.FailedTestsDetector.java
+----
+include::../strategies/failed/src/main/java/org/arquillian/smart/testing/strategies/failed/FailedTestsDetector.java[tag=documentation]
+----
+<1> Method that returns list of tests to execute
+<2> `TestSelection` requires fully qualified test class name and the strategy where is calculated
+
+=== Registering strategy
+
+Finally you need to register strategy in Smart Testing configuration file where you set the strategy name as key and the Maven coordinates as value.
+This file is required so *Smart Testing* can automatically register the artifacts in build tool script.
+
+As an example of registering previous strategy:
+
+[source, yml]
+.smart-testing.yml
+----
+customStrategies:
+    - smart.testing.strategy.failed=org.arquillian.smart.testing:strategy-failed:1.0.0
+----
+
+It is important to notice that the _key_ part must be prefixed with `const:mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/dependencies/StrategyDependencyResolver.java[name="SMART_TESTING_STRATEGY_PREFIX"]`.
+
+
+=== Alternative Approach
+
+You can still use another approach to register the pair _strategy name_:_Maven coordinates without configuring them in Smart Testing configuration file and it is using a System property.
+
+For example <<custom-strategy, previous example>> would be equivalent as running *Smart Testing* with next System property:
+
+`-Dsmart.testing.strategy.failed=org.arquillian.smart.testing:strategy-failed:1.0.0`
+
+Notice again that it must be prefixed with `const:mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/dependencies/StrategyDependencyResolver.java[name="SMART_TESTING_STRATEGY_PREFIX"]`.
+
+
```

Comment:
What I would like to see in this doc is also some hints about optimizations (calculating upfront before the build starts on the extension side). Something we did for Git resolutions as well as for Failed strategy. This is somewhat a hidden gem, but crucial to have a pleasant experience using our tool (with low overhead).